sdk-python/copilotkit/copilotkit_lg_middleware.py

```
@@ -0,0 +1,197 @@
+    # Private state for CopilotKit middleware
+    copilotkit: NotRequired[dict[str, Any]]
+
+
+class CopilotKitMiddleware(AgentMiddleware[AGUIState, Any]):
+    """AG-UI Middleware for LangGraph agents.
+
+    Handles frontend tool injection and interception for AG-UI protocol.
+    """
+
+    state_schema = AGUIState
+    tools = []
```

Ruff (0.14.10)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)